src/ini.c

```diff
@@ -0,0 +1,154 @@
+
+#include "ini.h"
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+
+#define INI_STRING_DELIM '"'
+#define INI_REM_CHAR1 '#'
+#define INI_REM_CHAR2 ';'
+#define INI_SECTION_START_CHAR '['
+#define INI_SECTION_END_CHAR ']'
+#define INI_EQU_CHAR1 '='
+#define INI_EQU_CHAR2 ':'
+
+#define _AT_EOL(p)           (*(p) == '\n')
+#define _AT_REMARK(p)        (*(p) == INI_REM_CHAR1 || *(p) == INI_REM_CHAR2)
+#define _AT_EQUALS(p)        (*(p) == INI_EQU_CHAR1 || *(p) == INI_EQU_CHAR2)
+#define _AT_SECTION_START(p) (*(p) == INI_SECTION_START_CHAR)
+#define _AT_SECTION_END(p)   (*(p) == INI_SECTION_END_CHAR)
+
+#define INI_SECTION_LEN (200)
+#define INI_LINE_LEN   (4096)
+
+
+char *_skip_space(char *ptr)
+{
+  while(ptr && *ptr && isspace(*ptr) && !_AT_EOL(ptr)) { ptr++; }
+  return ptr;
+}
+
+
+int _parse_section_header(char **parse_pos_ptr, char *section)
```

Comment:
strtok() has only very limited use. This parser doesn't only want to collect tokens, but also decide what and how to parse the next bit based on the type of delimiter(s) we encounter. What doesn't help that strtok() doesn't supply feedback on what delimiter it found and also destroys the original string content, so there's no easy backtracking if you didn't get the type of token you expected.

Reply:
I would still like to see some standard library use over hand-rolled loops. Perhaps strchr? 

And you say that you don't want to use strtok because it replaces the delimiter with \0, but _parse_section_header does the same thing. So what is the issue there?

Reply:
Resolution: Accept current implementation, but we are still open for improvements and/o replacement of the reader function(s) by a well supported library.